pinot-common/src/main/java/org/apache/pinot/common/utils/RoundRobinURIProvider.java

```diff
@@ -38,18 +41,30 @@ public class RoundRobinURIProvider {
 
   public RoundRobinURIProvider(URI originalUri)
       throws UnknownHostException, URISyntaxException {
-    String hostName = originalUri.getHost();
-    if (InetAddresses.isInetAddress(hostName)) {
+    this(originalUri, true);
+  }
+
+  public RoundRobinURIProvider(URI originalUri, boolean resolveHost)
+      throws UnknownHostException, URISyntaxException {
+    if (resolveHost) {
+      _uris = resolveHostToIPAddresses(originalUri);
+    } else {
       _uris = new URI[]{originalUri};
+    }
+    _index = new Random().nextInt(_uris.length);
+  }
+
+  public RoundRobinURIProvider(List<URI> originalUris)
+      throws UnknownHostException, URISyntaxException {
+    this(originalUris, true);
+  }
+
+  public RoundRobinURIProvider(List<URI> originalUris, boolean resolveHost)
```

Comment:
opinionated: I think keeping only 1 ctor (this one) would be cleaner. Callers can pass in a singleton list where required.